controllers/userController.js

```diff
@@ -16,6 +17,51 @@ exports.getAllUsers = async (req, res) => {
   }
 }
 
+exports.getUsers = async (req, res) => {
+
+  //get the current user id from the token
+  const token = getEvenToken(req)
+  const decoded = decodeTkn(token, req.config.JWT_KEY)
+  const id = decoded[constants.idClaim]
+
+  //check if the current user is an admin
+  let isAdmin = req.user.admin
+  
+  if(!isAdmin) {
+    //get the data of the current user
+    try {
+      const userData = await req.db.User.findById(id)
+
+      return res.status(HttpStatus.OK).json({
+        success: true,
+        userData
```

Comment:
we should return `user` instead of `userData`

Reply:
If we return req.user, should we still do:
`await req.db.User.findById(ObjectId(req.user._id))`
or is it ok to just return req.user if the user is not an admin?
`if(!isAdmin)
return res.status(HttpStatus.OK).json({
        success: true,
        req.user
      })`
But when the user updates his/her data does a middleware update the token?